\documentclass[11pt]{article}

\usepackage[margin=1.0in]{geometry}
\linespread{1.5}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage{cite}




\renewcommand{\bottomfraction}{.9}
\renewcommand{\topfraction}{.9}
\renewcommand{\textfraction}{0.1}
\renewcommand{\floatpagefraction}{.9}


\begin{document}

\section*{Introduction, Rough}

Over the years, a variety of models have been proposed to describe the effects of natural selection on protein-coding sequences, in a phylogenetic context. Traditionally, the focus has been on mechanistic codon-substitution models (see ref.~\cite{Anisimova2009} for a comprehensive review). Since their introduction in the 1990s, these models have seen great success in inferring protein evolutionary rates, or the nonsynonymous/synonymous rate ratio ($dN/dS$). This metric indicates how quickly a protein's constituent amino acids change \cite{GoldmanYang1994, MuseGaut1994, NielsenYang1998}, allowing for the identification of positively-selected regions in protein sequences \cite{ NielsenYang1998,Yangetal2000}. 

More recently, a second class of models, known as mutation-selection-balance (MutSel) models, has emerged as a popular alternative to $dN/dS$ models. Unlike mechanistic codon models, MutSel models explicitly model the dynamic balance between mutation and selection, rather than merely the final outcome (e.g. substitution) of this process \cite{HalpernBruno1998, YangNielsen2008, Rodrigueetal2010, Tamurietal2012}. Moreover, these models yield estimates of amino acid selection coefficients, which indicate the extent to which natural selection favors, or disfavors, particular amino acids at protein positions. These selection coefficients, which can in turn be scaled relative to a focal amino acid, the primary parameters of interest that MutSel models produce. Although MutSel models were first introduced over 15 years ago \cite{HalpernBruno1998}, they have seen virtually no use due to their high computational expense. However, recently, several computationally tractable model implementations have emerged \cite{RodrigueLartillot2014,Tamurietal2014}, allowing for the first time the potential for widespread use. 

Some have argued that MutSel models are more robust than $dN/dS$ and can better describe the evo process owing to their treatment of amino acid identities. More fine-grained modeling results than a $dN/dS$ analysis would yield. However, it is virtually unknown how these models really relate, so it remains unclear whether one model should be preferred over another. We don't know how parameter estimates even relate.

Although both $dN/dS$ and MutSel models describe the same fundamental process of protein evolution along a phylogeny, the relationship between these models is largely unknown. These two classes of models have largely been developed independently, and as a consequence we do not know whether parameters estimated from a $dN/dS$ model are similar, distinct, or even contradictory to those estimated from from a MutSel model of coding sequence evolution. Here, we aim to formalize the relationship between $dN/dS$ and MutSel models by examining the extent to which their focal parameters, $dN/dS$ and scaled amino acid selection coefficients, yield overlapping information about the evolutionary process. To this end, we derive a mathematical relationship between these two parameter classes, and we demonstrate that MutSel models fully embody the $dN/dS$ values. Using a simulation approach, we show that we can accurately estimate $dN/dS$ values using MutSel model parameter estimates, and these estimates correspond precisely to those inferred using a traditional $dN/dS$ maximum likelihood inference approach.  Importantly, we additionally show that this relationship holds only under regimes of purifying selection or neutral evolution ($dN/dS \leq 1$). Therefore, MutSel models are inherently unable to describe protein evolution under a regime of positive selection, in which $dN/dS > 1$. This result has important implications for circumstances under which MutSel model use is justified.

 
\section*{Mathematical relationship between selection coefficients and omega}
	
We describe here how to calculate $dN/dS$ from the parameters of a MutSel model. We assume the following: (i) the mutational process is symmetric, such that $\mu_{xy}=\mu_{yx}$ for all nucleotide pairs $xy$; (ii) all synonymous codons for a given amino acid have the same fitness; there is no synonymous rate variation or codon bias.

In the framework of a MutSel model, we can write the steady-state frequency of codon $i$ as
\begin{equation}\label{eq:fi}
 f_i=e^{s_i}\Big/\sum_k e^{s_k}\,,
\end{equation}
where the sum in the denominator runs over all 61 sense codons \cite{SellaHirsh2005}. Here, $s_i$ is the \emph{scaled selection coefficient} for codon $i$; larger $s_i$ correspond to higher frequencies of codon $i$. The fixation probability for a mutation from $i$ to $j$ is \cite{HalpernBruno1998,SellaHirsh2005}
\begin{equation}\label{eq:pi}
  \pi_{i\rightarrow j} = \frac{1-(f_i/f_j)^{1/N_e}}{1-f_i/f_j}
  \approx \frac{1}{N_e} \frac{\ln f_j - \ln f_i}{1-f_i/f_j}\,,
\end{equation}
where $N_e$ is the effective population size. We can calculate an evolutionary rate by summing over all fixation probabilities weighted by the frequency of the originating codon. For example, we can write the synonymous rate $K_\text{S}$ as
\begin{equation}\label{eq:KS}
  K_\text{S} = N_e \sum_i \sum_{j \in {\cal S}_i} f_i  \pi_{i\rightarrow j}\mu_{ij}\,,
\end{equation}
where ${\cal S}_i$ is the set of codons that are synonymous to codon $i$ and differ from it by one nucleotide substitution. To normalize $K_\text{S}$, we divide it by the number of synonymous sites $L_\text{S}$, which we can calculate as 
\begin{equation}\label{eq:LS}
  L_\text{S} = \sum_i \sum_{j \in {\cal S}_i} f_i \,.
\end{equation}
Under the assumption that all synonymous codons have equal fitness (all synonymous mutations are neutral), we have $\pi_{i\rightarrow j}=1/N_e$ \cite{CrowKimura1970}, and thus we find for $dS$, the synonymous rate per synonymous site,
\begin{equation}\label{eq:dS}
  dS = \frac{K_\text{S}}{L_\text{S}}= \frac{\sum_i \sum_{j \in {\cal S}_i} f_i\mu_{ij}}{\sum_i \sum_{j \in {\cal S}_i} f_i}.
\end{equation}
Similarly, for $dN$, the non-synonymous rate per non-synonymous site, we find
\begin{equation}\label{eq:dN}
  dN = \frac{K_\text{N}}{L_\text{N}}=\frac{ N_e \sum_i \sum_{j \in {\cal N}_i} f_i  \pi_{i\rightarrow j}\mu_{ij}}{\sum_i \sum_{j \in {\cal N}_i} f_i}\,,
\end{equation}
where ${\cal N}_i$ is the set of codons that are not synonymous to codon $i$ and differ from it by one nucleotide substitution. The quantities $K_\text{N}$ and $L_\text{N}$ are defined as in Eqs.~\eqref{eq:KS} and \eqref{eq:LS} but summing over $j\in {\cal N}_i$ instead of $j\in {\cal S}_i$.

Equations \eqref{eq:fi}--\eqref{eq:dN} establish a connection between the scaled selection coefficients $s_i$ (i.e., the primary parameters of a MutSel model) and the evolutionary rate ratio $dN/dS$. 



\section*{Methods, INCREDIBLY ROUGH}

\subsection*{Sequence simulation and omega inference}
We simulated protein-coding sequences as a continuous-time Markovprocess \cite{Yang2006} according to the MutSel model proposed by \cite{HalpernBruno1998}. This model's instantaneous rate matrix $Q = q_{ij}$, which describes the probability of substitution of from codon $i$ to codon $j$, is given by 

\begin{equation}
Q_{ij} = \left\{ \begin{array}{rl}
              0                                           &\mbox{multiple nucleotide changes} \\
              \mu_{ij}f_{ij}                          &\mbox{single nucleotide transversion} \\
              \kappa\mu_{ij}f_{ij}               &\mbox{single nucleotide transition} \\
         \end{array} \right.,
\end{equation} where $\mu_{ij}$ is the symmetric nucleotide mutation rate and $f_{ij}$ is the fixation probability from codon $i$ to $j$. The fixation probability is defined as \begin{equation}f_{ij} = ln\bigg{(}\frac{\pi_j\mu_{ij}}{\pi_i\mu_{ji}}\bigg{)}\bigg{/}\bigg{(}1 - \frac{\pi_i\mu_{ji}}{\pi_j\mu_{ij}}\bigg{)},\end{equation} where $\pi_i$ is the equilibrium frequency of codon $i$.

For each simulation, we derived steady-state amino acid, and hence codon, frequencies according to the Boltzmann distribution,
\begin{equation} F(a) = \frac{e^{s_a\beta}}{\sum_b e^{s_b\beta}} \end{equation}, where $F(a)$ is the equilibrium frequency of amino acid $a$, and the denominator sums over all 20 amino acids \cite{SellaHirsh2005, Ramseyetal2011}. In this framework, $s_a$ represents the scaled selection coefficient for amino acid $a$, analogous to the primary parameter given by a MutSel model. Unless otherwise stated, \texbf{$\beta$ was set to 2.0 for all simulations (I MADE THIS UP, COME BACK TO IT ?????)}. 
  
  
Once amino acid frequencies were determined, we assigned them to individual amino acids as follows. There was a certain number of values, out of 20, greater than 0.05 (the freq expected by random chance). For this number, we got a set of reasonably co-occurring amino acids based on mean pair-wise grantham scores. We selected a random group of aa's such that the mean distance was less than or equal to 100. The remaining frequencies were assigned randomly to the remaining amino acids.

Simulation parameters - 2 taxon tree with branch lengths fixed at 0.005. Mu, the symmetric base mutation rate was fixed at 1e-6, and we varied kappa from time to time. We evolved each position according to the same model; no site-wise variation.

For each simulated sequence set, we inferred $dN/dS$ values using two approaches: the derivation presented in this paper (see Results) and using the standard maximum likelihood GY94 model \cite{GoldmanYang1994} within the HyPhy package \cite{KosakovskyPondetal2005}. This model includes two primary parameters, $\omega$ and $\kappa$. For each inference, we fixed $\kappa$ to the known simulated value, and we provided HyPhy with equal equilibrium codon frequencies, such that each codon had a frequency of $1/61$. This was necessary to achieve accurate $dN/dS$ estimates, and is discussed more in depth in Results. Using ML, we estimated a single average $\omega$ value for each alignment. 



\section*{Results}

\subsection*{$dN/dS$ values fully encapsulated by scaled selection coefficients}

To validate our derived relationship between $dN/dS$ values and scaled selection coefficients, we simulated protein-coding sequences along a lineage according to the Halpern-Bruno mutation-selection model \cite{HalpernBruno1998}. For each simulation set, we fixed all mutation rates to $10^{-6}$, and we selected steady-state amino acid frequencies according to \begin{equation} F(a) = \frac{e^{s_a\beta}}{\sum_b e^{s_b\beta}} \end{equation}, where $F(a)$ corresponds to the frequency of amino acid $a$. These amino acid frequencies are analogous to the scaled selection coefficient parameters which a mutation-selection model would produce. Unless otherwise stated, we set $\beta = 2.5$. Higher values of $\beta$ indicate stronger constraint on the amino acid distributions; as $\beta$ approaches infinity, only a single amino acid will effectively have a non-zero frequency. 

Following simulation, we calculated a $dN/dS$ value using both standard ML methods, according to the GY94 \cite{GoldmanYang1994} model, and our relationship. Figure~\ref 
As shown in FigureX, the values nearly perfectly agree, demonstrating that all information about $dN/dS$ is fully encapsulated by the mutation-selection model parameters.


We can additionally show that GC content does not influence this relationship. A linear model with error ~ gc yields p=0.487, meaning that this relationship is robust to nucleotide content.

This relationship is also robust to mutational bias, specifically TI-TV bias, represented by the parameter kappa.


\subsection*{Maximum Likelihood inferences strongly biased by equilibrium frequency specifications}

An emerging result from this study was that specs really matter. In 
This is because of where selection is represented. In mutsel models, selection is essentially manifested by the frequencies, whereas in dnds models, it is represented by omega. If you provide true frequencies, or ask for it to be calculated from data, then the frequencies hold the effects of selection, causing inflated estimates of dnds. However, if you set all codons to 1/61, this scenario effectively represents a neutral case with no effects of selection, which then allows the omega parameter to capture the effects.






















\subsection*{Mutation-selection-balance models are only valid for purifying selection}

\textbf{I really have no idea how to write up proofs, so I've virtually just latex'd the mathematica document. But at least the equations are there!}

To show that mutation-selection models only corresponds to $dN/dS \leq 1$, we make use of that fact that each calculation of $dN/dS$, as described in equations \eqref{eq:fi}--\eqref{eq:dN}, entails summing the forward and backward fixation probabilities between codons, which are in turn divided by the codon frequency sums. We additionally assume that all mutation rates $\mu_{ij}$ are equal, and as all values for $N$ will ultimately cancel, we exclude them from the following proof. We will deal with the case of two nonsynonymous codons, $i$ and $j$, and we write their frequencies as $x$ and $y$, respectively.

As follows from \eqref{eq:pi}, the sum of the probabilities going from codon $i$ to codon $j$ and from codon $j$ to codon $i$ is \begin{equation} xP(x,y) + y(P(y,x) = \frac{2xy [\ln x - \ln y]}{x - y}\end{equation}, and we will demonstrate that this value is necessarily $\leq x + y$ for $x,y \geq 0$ and $x \leq y$.

To this end, we define the function \begin{equation} F(x,y) = x + y - \frac{2xy [\ln x - \ln y]}{x - y} \end{equation}. Thus, we show that $F(x,y) \geq 0$. For the condition $x=y$, this is straightforward to show, as $\lim_{x \to y}F(x,y) = 0$.
We now show that the first derivative of $F(x, y)$ is negative throughout $x \in (0,y)$, thus proving that $F(x, y)$ has to be monotonically decreasing, and hence $\geq0$, in this interval.
	
\newpage
	


	
 Several things emerge from this analysis - 
1. If you fit a mutsel model and calculate dnds from it, it's just as good as if you had used a mech codon model. 
2. we prove that dnds, when calc'd from mutsel, must be less than 1. although generally acknowledged that purifying selection is a feature, we demonstrate it precisely and reveal that, in cases of positive selection, mutsel models are likely not appropriate.
3. we also have a more realistic way to assess performance of dnds models. bias is introduced by codon frequencies used by the model. otherwise, freqs will likely capture selective pressure, rendering omega estimates bonkers. typically highly elevated. More work should be done in this field, for instance, in dnds analyses in which codon frequencies are somehow constrained (eg membrane protein in which hydrophobic enriched) may result in artifactually high dnds.
Discussion points:
- Important insight is that dN/dS>1 inherently cannot be described by mutation-selection models, and thus at those sites its results may be misleading. In particular, possibly a confounding factor in the Rodrigue implementation, as positively selected sites in the alignment could introduce bias.
- Importance of examining intersections between models. Must understand how estimates from one relate to another. Helps to ensure robust results; model agreement is key, so we must formulate explicit relationships among them to systematically assess agreement.
- 

	
	

Future directions:
1. Consider which info the models share and which they don't. For purifying selection, params a dNdS model would yield are fully contained within a mutsel model. On the other hand, the model overlap disappears under positive selection, when steady-state equilibrium is violated. Thus, mutsel models are sufficient for purifying, but not at all useful for positive selection.

dNdS limitations:
One such limitation is that the $dN/dS$ parameter ignores the influence of site-specific amino acid propensities.  It is universally recognized that a particular position in a protein will only tolerate certain amino acids, due either to functional or structural constraints. However, $dN/dS$ ignores this key aspect of protein evolution and considers all nonsynonymous changes, regardless of which amino acid was substituted, as having equal weight on protein fitness, a biologically implausible assumption. An additional limitation is that codon-substitution models merely describe the result of the evolutionary process, rather than the explicit underlying mechanism producing those results. More precisely, substitutions in protein-coding sequences result from an ongoing mutation-selection balance. When a mutation occurs in a DNA sequence, natural selection must act on this mutation, either by disfavoring it, resulting in the mutation's removal from the population, or favoring it, ultimately yielding an amino acid replacement or substitution. By overlooking this underlying mechanism and focusing only on whether substitutions have occurred, codon substitution models are unable to capture the full extent of the evolutionary process. 


	
\newpage
\bibliographystyle{plos2009}
\bibliography{bibliography}	


\bigskip

\begin{figure*}[H]
\centerline{\includegraphics[width=6in]{figures/gc_2beta.pdf}}
\caption{\label{kappa} GC content additionally does not influence the relationship.}
\end{figure*}

\begin{figure*}[H]
\centerline{\includegraphics[width=6in]{figures/kappa_1beta.pdf}}
\caption{\label{kappa} Value of kappa does not matter for agreement with derived and ML omegas. Relationship robust to differences in (symmetric) mutational spectrum.}
\end{figure*}

\begin{figure*}[H]
\centerline{\includegraphics[width=6in]{figures/freqspec_1beta.pdf}}
\caption{\label{freqspec_compare} Equilibrium codon frequency specification to ML inference matters. Omega estimates agree when specify equal codon frequencies, and error increases as frequency specifications are more and more tailored to the data, ultimately resulting in wildly inflated values when the real frequencies are used.}
\end{figure*}

\begin{figure*}[H]
\centerline{\includegraphics[width=6in]{figures/freqspec_entropy_error.pdf}}
\caption{\label{freqspec_compare} Issues with frequency specifications strongly related to the codon frequencies in the data set. Issue is more egregious when there are relatively few codons, based on entropy. As entropy increases (more permissive, and thus data set codon frequencies are flatter), the error decreases and ML more approximates the true omega value.}
\end{figure*}


\begin{figure*}[H]
\centerline{\includegraphics[width=6in]{figures/convergence.pdf}}
\caption{\label{conv} Convergence of derived $dN/dS$ and ML estimates of $dN/dS$. Each point represents results from a single simulation. The y-axis indicates relative error of the ML $dN/dS$ estimates, and the x-axis indicates sequence length on a log-scale. As the sequence length, or the data set size, increases, the two $dN/dS$ estimates converge to the same value. Note that this simulated data used a beta of 2.5.}
\end{figure*}


	
	
	
\end{document}

