
\documentclass[11pt]{article}
\usepackage[margin=1.0in]{geometry}

%\documentclass{pnas}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage{lscape}
\newcommand{\xline}[0]{\noindent\underline{\makebox[0.15cm][l]{}}}
\makeatletter
\def\hlinewd#1{%
	\noalign{\ifnum0=`}\fi\hrule \@height #1 \futurelet
	\reserved@a\@xhline}
\makeatother
\begin{document}
	
	
%\title{The relationship between scaled selection coefficients and dN/dS}
%	
%\author{Stephanie J. Spielman\affil{1}{Department of Integrative Biology, Center for Computational Biology and Bioinformatics, and Institute of Cellular and Molecular Biology.
%		The University of Texas at Austin, Austin, TX 78712, USA.} 
%	\and
%	Claus O. Wilke\affil{1}{}
%}
	
%\contributor{Submitted to Proceedings of the National Academy of Sciences of the United States of America}
%\maketitle

%\begin{article}

\section*{Significance Statement} 
Phylogenetic models of coding-sequence evolution provide rigorous and sophisticated tools in evolutionary studies. While various such frameworks exist, how these approaches relate remains an open question. Here, we elucidate the relationship between two competing frameworks that quantify selection strength, $dN/dS$ and selection coefficient models. In doing so, we illuminate previously unrecognized model properties and limitations. We show that MutSel models correspond strictly to regimes of purifying selection ($dN/dS < 1$), although selection acting on synonymous changes can produce spurious signals of positive selection. We additionally find that most $dN/dS$ models infer systematically biased parameter estimates. Moreover, model fit metrics (AIC) cannot necessarily identify which model produces the most accurate inferences, and can therefore be counterproductive and misleading in model selection. %120 words



\begin{abstract}
Computational methods discerning the strength of natural selection in protein-coding sequences assume a central role in comparative sequence analyses. Various such models have been proposed, yet how these independent approaches relate to one another, in particular whether inferences made with competing models yield comparable or distinct results, remains entirely unknown. In this study, we elucidate the relationship between two widely-used phylogenetic modeling frameworks: $dN/dS$ models and mutation-selection-balance (MutSel) models, which estimate amino acid and/or codon scaled selection coefficients. We derive a formal mathematical framework to calculate $dN/dS$ directly from selection coefficients, allowing us to gain unprecedented insight into the behaviors, limitations, and applicabilities of these competing models. In particular, we prove that MutSel models correspond strictly to $dN/dS \leq 1$, and thus they cannot accomodate positive, diversifying selection ($dN/dS > 1$). However, if synonymous changes are not neutral, we find that $dN/dS$ computed from selection coefficients can take on arbitrarily high values, even to infinity. This result broadly demonstrates that purifying selection on synonymous codons can erroneously indicate that positive selection is acting. We additionally find that $dN/dS$ models systematically underestimate $dN/dS$ from sequence data evolved under an asymmetric mutation model. Strikingly, the best-fitting $dN/dS$ model parameterization, based on AIC, was never the model which yielded the least biased and most precise $dN/dS$ estimates. Thus, selecting models based solely on fit can be counterproductive and positively misleading. We conclude that establishing mathematical links among modeling frameworks represents a novel, powerful strategy to pinpoint previously unrecognized model limitations and strengths. %249 words
\end{abstract}


%\keywords{dN/dS | mutation-selection-balance models | scaled selection coefficients | Markov models of sequence evolution}


\section*{Introduction}
		
The oldest and most-widely used method to infer selection pressure in protein-coding genes calculates the evolutionary rate ratio $dN/dS$, which represents the ratio of non-synonymous to synonymous substitution rates. This metric indicates how quickly a protein's constituent amino acids change, relative to synonymous changes, and it is commonly used to identify protein sites that experience purifying selection ($dN/dS<1$), evolve neutrally ($dN/dS\approx1$), or experience positive, diversifying selection ($dN/dS>1$) \cite{NielsenYang1998, Yangetal2000, KosakovskyPondFrost2005b, Huelsenbecketal2006}. In phylogenetic contexts, $dN/dS$ is typically calculated using a maximum likelihood (ML) approach \cite{GoldmanYang1994,MuseGaut1994,NielsenYang1998,Yang2006}. ML methods assume a continuous time Markov model of sequence evolution and have become a staple of comparative sequence analysis since their introduction in the 1990s (see ref \cite{Anisimova2009} for a comprehensive review). Throughout this paper, we will refer to these models as $dN/dS$-based models.

A second class of Markov models, known as mutation-selection-balance (MutSel) models, are increasingly being viewed as a viable alternative to the $dN/dS$ framework. While $dN/dS$-based models describe how quickly a protein's constituent amino acids change, MutSel models assess the strength of natural selection acting on specific mutations. Couched firmly in population-genetic theory, the MutSel framework estimates site-specific scaled selection coefficients $S=2N_es$, which indicate the extent to which natural selection favors, or disfavors, particular codon and/or amino acid changes \cite{HalpernBruno1998,YangNielsen2008,Rodrigueetal2010,Tamurietal2012}. Although first introduced over 15 years ago \cite{HalpernBruno1998}, MutSel models have seen little use due to their high computational expense. Recently, however, several computationally tractable model implementations have emerged \cite{RodrigueLartillot2014,Tamurietal2014}, allowing for the first time the potential for widespread adoption.		
		
Over the course of twenty years developement, $dN/dS$-based models have advanced to a high level of sophistication. These models can accommodate a variety of evolutionary scenarios, including synonymous rate variation \cite{MuseGaut1994,KosakovskyPondMuse2005} and episodic \cite{KosakovskyPondetal2011,MEME} and/or lineage-specific selection \cite{YangNielsen2002,Zhangetal2005,KosakovskyPondFrost2005a}, and they can also incorporate information regarding protein structure and epistatic interactions \cite{Robinsonetal2003,Thorneetal2007,Rodrigueetal2009,Scherreretal2012,MeyerWilke2012}. This flexibility, along with accessible software implementations \cite{KosakovskyPondetal2005,Yang2007,Delport2010}, makes $dN/dS$-based models an attractive analysis choice. On the other hand, some have argued that MutSel models, given their explicit basis in population-genetics theory and attention to site-specific amino-acid fitness differences, offer a more mechanistically realistic approach to studying coding-sequence evolution \cite{HalpernBruno1998,Rodrigueetal2010,Tamurietal2012,Thorne2012}. Moreover, a growing body of literature has demonstrated that $dN/dS$ estimates are particularly sensitive to violations in model assumptions, calling into question the general utility of $dN/dS$-based models \cite{Rochaetal2006,KryazhimskiyPlotkin2008,Mugaletal2014}.
		
Although both MutSel and $dN/dS$-based models describe the same fundamental process of coding-sequence evolution along a phylogeny, it is unknown how these two modeling frameworks relate to one another. In particular, as these inference methods have been developed independently, it remains an open question whether or not parameter estimates from one model are comparable to those of the other model. As a consequence, although certain rhetorical arguments may be made in favor of using one method over another, there is currently no formalized, concrete rationale to guide researchers in their methodological choices. Elucidating the relationship between these competing modeling frameworks will more precisely reveal under which circumstances the use of these models is justified and has great potential to reveal previously unrecognized model properties and/or limitations.
		
Here, we formalize the relationship between these two modeling frameworks by examining the extent to which their respective focal parameters, $dN/dS$ and scaled selection coefficients, yield overlapping information about the evolutionary process. To this end, we derived a mathematical relationship between $dN/dS$ and scaled selection coefficients. We find that $dN/dS$ values can be precisely calculated from scaled selection coefficients, and that $dN/dS$ accurately captures the selective pressures indicated by a given distribution of scaled selection coefficients. Furthermore, we have proven that, when synonymous mutations are neutral, $dN/dS$ calculated from selection coefficients is necessarily less than 1. This proof demonstrates that MutSel models are inherently only able to model purifying selection, and therefore would be an inappropriate model choice if positive selection is expected. However, when synonymous codons have different fitnesses, it is possible to recover $dN/dS$ values above 1, even though no positive selection is occurring. 

Finally, this relationship provides a uniquely rigorous platform to examine the performance of $dN/dS$-based models. Typically, researchers evaluate performance of a given inference framework through simulations that adhere to the underlying model's assumptions (with a notable exception of ref.\ \cite{Holder2008}). Indeed, simulated data is usually generated according to the same model as the inference framework, allowing for a direct comparison between the true and estimated parameter values. While this strategy is critical for testing whether a model implementation behaves as expected, it inherently cannot assess model performance when the data arises from a different mechanism than that of the inference framework, as is obviously the case in real sequence analysis. A more sensitive test of model performance would examine how a given method performs when data is simulated under entirely different conditions. Unfortunately, such an approach is typically infeasible because true parameter values would be unknown, and thus model performance would remain untested. 

The relationship we have established between $dN/dS$ and selection coefficients allows us to overcome this limitation, as we can determine the true $dN/dS$ value directly from MutSel model parameters. Thus, we can assess performance of $dN/dS$-based inference frameworks by simulating data with a MutSel model and then comparing inferred $dN/dS$ ML estimates (MLEs) to $dN/dS$ values computed from selection coefficients. Using this strategy, we find that, in the absence of mutational bias, $dN/dS$ values inferred in an ML framework agreed precisely with those calculated from scaled selection coefficients. However, as mutational bias increases, $dN/dS$ MLEs become increasingly biased away from their true values, even under a variety of ML model parameterizations. Surprisingly, the best-performing ML model parameterizations are not those which exhibited the best fit to the data (measured by AIC), ultimately revealing that relying on model fit as a litmus-test for model performance can be an ineffective and misleading strategy. 

		
\section*{Results and Discussion}
		
		
\subsection*{Theoretical model.}
This section contains a rederivation of results presented in refs.\ \cite{HalpernBruno1998,YangNielsen2008}, reproduced here to place the remainder of our work into context. We model sequence evolution using the Halpern-Bruno MutSel modeling framework under the assumptions of a fixed effective population size $N_e$ and constant selection pressure over time \cite{HalpernBruno1998,YangNielsen2008,Tamurietal2012,Thorne2012}.  This continuous-time reversible Markov process is governed by the $61 \times 61$ transition matrix $M(t) = e^{Qt}$, where the matrix $Q = q_{ij}$ gives the instantaneous substitution probabilities between all 61 sense codons, and diagonal elements of $Q$ satisfy $q_{ii} = -\sum_{i \neq j} q_{ij}$. We assume that only single-nucleotide substitutions occur instantaneously. 

Let $f^\text{codon}_i$ be the fitness of codon $i$, and let the selection coefficient acting on a mutation from codon $i$ to codon $j$ be $s_{ij} = f^\text{codon}_j - f^\text{codon}_i$ \cite{SellaHirsh2005,YangNielsen2008}. The fixation probability for this mutation is \cite{Kimura1962,HalpernBruno1998,YangNielsen2008}
\begin{equation}\label{eq:u_ij}
u_{ij} \approx \frac{2s_{ij}}{1 - e^{-2N_es_{ij}}} = \frac{1}{N_e}\frac{2N_es_{ij}}{1 - e^{-2N_es_{ij}}} .
\end{equation}. We further define $S_{ij} = 2N_es_{ij}$ as the scaled selection coefficient for this change \cite{YangNielsen2008}. The probability of a substitution from codon $i$ to $j$ is therefore
\begin{equation}\label{eq:q_ij}
q_{ij} = N_em_{ij}u_{ij} = m_{ij}\frac{S_{ij}}{1 - e^{-S_{ij}}} , 
\end{equation} where $m_{ij}$ is the codon mutation rate, which represents the rate at which codon $i$ transitions to codon $j$  \cite{HalpernBruno1998,SellaHirsh2005}. If we assume that $m_{ij}$ only describes single-nucleotide changes, it can be written as the nucleotide mutation rate $\mu_{s_it_j}$, where $s_i$ is the source nucleotide in codon $i$, and $t_j$ is the target nucleotide in codon $j$.

We now show how $S_{ij}$ can be written in terms of mutation rates and stationary (equilibrium) codon frequencies $P_i$. As this system statisfies detailed balance (reversibility) \cite{HalpernBruno1998}, we have 
\begin{equation}\label{eq:DB}
q_{ij}P_i = q_{ji}P_j .
\end{equation} From equations \eqref{eq:q_ij} and \eqref{eq:DB}, we can write the ratio of substitution probabilities as 
\begin{equation}\label{ratio_q_ij}
\frac{P_i}{P_j} = \frac{m_{ji} S_{ji} (1-e^{-S_{ij}})} {m_{ij} S_{ij} (1-e^{-S_{ji}})} .
\end{equation} Using $S_{ij} = -S_{ji}$, we find that
\begin{equation}\label{eq:s_pmu}
S_{ij} = \ln\bigg{(} \frac{P_jm_{ji}}{P_im_{ij}} \bigg{)} .
\end{equation} This equation, previously derived in ref.\ \cite{HalpernBruno1998}, establishes a relationship between scaled selection coefficients and the stationary codon frequencies of the Markov chain. Moreover, in the specific case of symmetric mutation rates $m_{ij} = m_{ji}$, we have $S_{ij} = \ln\big{(}P_j/P_i\big{)}$ \cite{SellaHirsh2005}. 


		
\subsection*{Mathematical relationship between scaled selection coefficients and \emph{dN/dS}.} 

Using the theory laid out in the previous subsection, we derive respective expressions for nonsynonymous and synonymous evolutionary rates, which we can divide to obtain the evolutionary rate ratio $dN/dS$. We write the nonsynonymous rate $K_\text{N}$ as 
\begin{equation}\label{eq:KN}
	K_\text{N} = \sum_i \sum_{j \in {\cal N}_i} P_i q_{ij} \,,
\end{equation}
where ${\cal N}_i$ is the set of codons that are nonsynonymous to codon $i$ and differ from it by one nucleotide, and the substitution probability $q_ij$ is defined in equation \eqref{eq:q_ij}. To normalize $K_\text{N}$, we divide it by the number of nonsynonymous sites, which we calculate according to the mutational opportunity definition of a site \cite{GoldmanYang1994, Yang2006} as 
\begin{equation}\label{eq:LN}
	L_\text{N} = \sum_i \sum_{j \in {\cal N}_i} P_i m_{ij}\, .
\end{equation} Thus, we find that 
\begin{equation}\label{eq:dN}
	dN = \frac{K_\text{N}}{L_\text{N}}=\frac{\sum_i \sum_{j \in {\cal N}_i} P_i q_{ij}} {\sum_i \sum_{j \in {\cal N}_i}P_i m_{ij}}\,.
\end{equation} Similarly, for $dS$, the synonymous evolutionary rate $K_\text{S}$ per synonymous site $L_\text{S}$, we find
\begin{equation}\label{eq:dS}
	dS = \frac{K_\text{S}}{L_\text{S}}=\frac{\sum_i \sum_{j \in {\cal S}_i} P_i q_{ij} } {\sum_i \sum_{j \in {\cal S}_i} P_i m_{ij} }\,,
\end{equation}
where ${\cal S}_i$ is the set of codons that are synonymous to codon $i$ and differ from it by one nucleotide substitution. The quantities $K_\text{S}$ and $L_\text{S}$ are defined as in Eqs.~\eqref{eq:KN} and \eqref{eq:LN} but sum over $j\in {\cal S}_i$ instead of $j\in {\cal N}_i$. Moreover, if we assume that mutation rates are symmetric and that all synonymous codons have equal fitness (i.e.\ synonymous mutations are neutral), the synonymous fixation rate satisfies $u_{ij | j \in {\cal S}_i} = 1/N_e$ \cite{CrowKimura1970}, and hence the substitution probability becomes $q_{ij} = m_{ij}$. In this circumstance, the value for $dS$ reduces to 1.
		
				
\subsection*{MutSel models strictly describe purifying selection.}

We examined the relationship between $dN/dS$ and scaled selection coefficients by simulating 200 distributions of amino acid scaled fitness values, $F^\text{aa}_a = 2Nf^\text{aa}_a$, from a normal distribution $\mathcal{N}(0,\sigma^2)$, where a unique $\sigma^2$ for each fitness distribution was drawn from a uniform distribution $\mathcal{U}(0,4)$. Higher values for $\sigma^2$ correspond to larger fitness differences among amino acids, causing selection to act more strongly against nonsynonymous changes. Thus, higher $\sigma^2$ values indicate strong purifying selection, low values indicate weaker purifying selection, and finally $\sigma^2 = 0$ indicates that all amino acids are equally fit. We note that these $F^\text{aa}_a$ quantities correspond exactly to the amino-acid propensity parameters estimated by currently available site-specific MutSel inference methods \cite{RodrigueLartillot2014,Tamurietal2014}.

We then converted each amino-acid fitnesses distribution to a corresponding set of codon fitnesses, as described in \emph{Methods}. Briefly, for 100 of the distributions, we assumed that all synonymous codons had the same fitness, but for the other 100 distributions we allowed synonymous codons to have different fitnesses. Using equations \eqref{eq:KN} - \eqref{eq:dS}, we computed $dN/dS$ for each distribution of codon fitnesses. For these calculations, we assumed the symmetric mutation model HKY85 \cite{HKY85}, which is specified by the parameters $\mu$, the nucleotide mutation rate, and $\kappa$, the ratio of transititions to tranversions. Specifically, transititions occur at a rate $\mu\kappa$, and tranversions at a rate $\mu$. We used $\mu = 10^{-6}$ for all simulations, while we selected a unique value for $\kappa$ for each simulation from $\mathcal{U}(1,6)$.

Under neutral evolution, we expect that $dN/dS = 1$, and as purifying selection increases in strength, $dN/dS$ should correspondingly decrease. Therefore, we expect that $dN/dS$ will decline with the variance ($\sigma^2$) of the distribution of amino acid fitness values. Indeed, we observed a strong, negatively correlation between these quantities. When fitness differences among amino acids were very high, $dN/dS$ took on lower values, properly reflecting stronger purifying selection (Figure~\ref{dnds_variance}). This correlation was much stronger for fitness distributions without synonymous selection (Figure~\ref{dnds_variance}A) than for those with synonymous selection (Figure~\ref{dnds_variance}B). This difference emerged because fitness differences among synonymous codons obscured underlying amino-acid fitness differences. Even so, selection on synonymous codons did not negate the significant correlation between $dN/dS$ and overall selection strength.

Importantly, Figure~\ref{dnds_variance}A demonstrates that, in the limiting case when $\sigma^2$ approaches 0, and thus all codons have virtually the same fitness, $dN/dS$ converges to 1. In other words, when the protein-coding sequence evolved neutrally, selection coefficients correctly yielded a $dN/dS \approx 1$. Furthermore, we never recovered $dN/dS > 1$ when synonymous changes were neutral. This finding reveals an important property of MutSel models: they inherently cannot describe positive, diversifying selection. Indeed, in Appendix 1, we have proven that scaled selection coefficients strictly yield $dN/dS \leq 1$, under the assumptions that synonymous changes are neutral and nucleotide mutation is unbiased. This proof formalizes the MutSel model's underlying assumption that selection pressure is constant over the phylogeny, and thus the protein evolves under equilibrium conditions. Although this proof assume symmetric nucleotide mutation rates, we do not expect that deviations from this assumption will have dramatic effects on $dN/dS$ estimates. 

\subsection*{Purifying selection on synonymous changes can produce \emph{dN/dS}$>$1.}

The restriction $dN/dS \leq 1$ does not hold when synonymous changes are not neutral, as seen in Figure~\ref{dnds_variance}B. In other words, even though the Halpern-Bruno explicitly assumes that the system is at equilibrium \cite{HalpernBruno1998,Thorne2012}, $dN/dS$ can readily be greater than 1. In fact, it is theoretically possible to achieve arbitrarily high $dN/dS$ values when synonymous codon substitutions carry fitness changes. In the most extreme case of synonymous selection, where only a single codon per amino acid is selectively tolerated, the number of synonymous changes $K_\text{S} = 0$, and thus the value for $dN/dS$ approaches infinity. Given that the MutSel model framework assumes an overarching regime of purifying selection, this finding might seem paradoxical. However, the logical argument that $dN/dS > 1$ represents positive, diversifying selection assumes that the rate of synonymous change may be used as a neutral benchmark, an assumption clearly violated when selection acts on synonymous changes. Thus, in theory, what is classically termed positive selection can result simply from strong synonymous fitness differences. 

That sequences evolving under purifying selection can spuriously bear the hallmark of positive, diversifying selection highlights the pitfalls of naively interpreting $dN/dS$ values. Indeed, evolutionary constraints which induce synonymous selection are pervasive and affect virtually all domains of life \cite{GuZhouWilke2010}, from viruses \cite{Cuevasetal2011,ZaniniNeher2013} to plants \cite{Guetal2012} to Metazoa \cite{Duret2002,Chamaryetal2006,HershbergPetrov2008,PlotkinKudla2010,Lawrieetal2013}. Particular genomic regions with strong synonymous selection include exonic splicing enhancers \cite{Schattner2006,Parmleyetal2006,ParmleyHurst2007}, RNA secondary structures \cite{Chamaryetal2005,Schattner2006,Cuevasetal2011,ZaniniNeher2013}, and sites near translation initiation \cite{GuZhouWilke2010}. In addition, both selection against protein misfolding and for translation efficiency tend to induce synonymous selection in a gene-specific manner \cite{WillifordDemuth2012,Agasheetal2013}, most notably in highly expressed genes \cite{DrummondWilke2008,Lawrieetal2013}. While synonymous selection may not dominate genomes in organisms with relatively small effective population sizes \cite{Chamaryetal2006, PlotkinKudla2010}, it certainly acts strongly at specific sites and/or small, local regions. As $dN/dS$ ratios are typically measured on a per-site basis, we expect that some sites with $dN/dS > 1$ may in fact be false positives in the detection of positive, diversifying selection. We offer several approaches to ease this concern in \emph{Conclusions}.



\subsection*{Relationship between \emph{dN/dS} and scaled selection coefficients provides a novel benchmarking approach.}

The relationship we have established between $dN/dS$ and scaled selection coefficients offers a unique opportunity to assess the robustness of $dN/dS$-based inference methods. It is conventional practice in model development to benchmark models against data simulated according to the model itself. While crucial for testing whether a given model has been correctly implemented, this strategy inherently cannot discern how the model behaves when data arose from a different mechanistic process. To overcome this limitation, we applied a novel benchmarking approach which used the theoretical relationship among modeling frameworks to assess the accuracy and specific utility of those models. This approach, outlined in Figure~\ref{reg_conv}A, entailed comparing $dN/dS$ values calculated from selection coefficients to those inferred by an $dN/dS$-based model. In addition, because MutSel models are based explicitly on population genetics theory, these simulated alignments are likely far more similar to real sequence data than are alignments simulated under a $dN/dS$-based model, and therefore provide excellent benchmarking data.

Using the selection coefficients and symmetric mutation rates from the previous two subsections, we simulated alignments using standard methods \cite{Yang2006} according the Halpern-Bruno MutSel model \cite{HalpernBruno1998}. We then inferred a $dN/dS$ value for each alignment using the GY94 matrix \cite{GoldmanYang1994,NielsenYang1998}, which estimates $dN/dS$ by obtaining a maximum likelihood estimate (MLE) for the parameter $\omega$. Throughout the remaining text, we refer to $dN/dS$ inferred using ML as $\omega$, and to $dN/dS$ computed using equations \eqref{eq:KN} - \eqref{eq:dS} simply as $dN/dS$. 

We found that $dN/dS$ values agree nearly perfectly with $\omega$ MLEs (Figure~\ref{reg_conv}B), and indeed this relationship was robust to both synonymous selection and uneven nucleotide composition (simulated alignments featured GC contents ranging from 0.21-0.89). Additionally, Figure~\ref{reg_conv}C demonstrates that $\omega$ converged to the true $dN/dS$ value as the size of the data set, represented by simulated alignment length, increased. These results unequivocally showed that the $dN/dS$ quantity is fully contained within MutSel model parameters. When nucleotide mutation is symmetric,  $dN/dS$-based model-inference methods behave exactly as expected and yield precise $dN/dS$ estimates. This finding has important implications for modeling choices; although the MutSel framework might model the sequence evolution in a way that more mechanistically matches the evolutionary process, $dN/dS$-based models may suffice to model selective forces in phylogenetic data.


\subsection*{Biased \emph{dN/dS} estimates under asymmetric mutation models.}
We next sought to test the accuracy of $dN/dS$-based models using more realistic parameter values. To this end, we determined codon fitness distributions from 498 unique distributions of experimentally-derived, site-specific amino-acid fitnesses for H3N2 influenza nucleoprotein (NP), given by ref. \cite{Bloom2014a}. We combined each of these fitness distributions with three sets of experimentally-determined mutation rates, either for NP \cite{Bloom2014a}, yeast \cite{Zhu2014}, or polio virus \cite{Acevedo2014}, to determine $498 \times 3 = 1494$ distinct distributions of steady-state codon frequencies (see \emph{Methods} section for details). While all three mutation matrices were asymmetric, each featured a differing degree of mutational bias; specifically, the mean ratios $\mu_{xy} / \mu_{yx}$ for NP, yeast, and polio mutation rates are 1.03, 1.69, and 5.25, respectively. For each resulting set of stationary codon frequencies, in combination with its respective set of mutation rates, we calculated $dN/dS$ and simulated alignments from which we inferred $\omega$. Note that we assumed no selection on synonymous codons for these calculations.

$dN/dS$-based model matrices account for nucleotide mutational bias by incorporating either target codon \cite{GoldmanYang1994} or target nucleotide \cite{MuseGaut1994} frequencies; these frameworks are known, respectively, as GY-style and MG-style models \cite{KosakovskyPond2010}. For example, the instantaneous rate matrix element giving the substitution probability from codon AAA to AAG would contain the target codon frequency $P_\text{AAG}$ in GY-style models but the target nucleotide frequency $\pi_\text{G}$ in MG-style models. Although the GY-style and MG-style matrices follow different forms, it is possible to write MG-style matrices such that they conform to the reversible GY-style framework, as we show in Appendix 2.

Previous works have suggested that MG-style and GY-style models yield different $\omega$ estimates \cite{KosakovskyPondMuse2005,Yap2010}, so we inferred $\omega$ according to both GY- and MG-style frameworks. For GY-style models, we used the frequency estimators F61 \cite{GoldmanYang1994}, F3x4 \cite{GoldmanYang1994}, CF3x4 \cite{KosakovskyPond2010}, and F1x4 \cite{MuseGaut1994}. For MG-style models, we considered both a parameterization in which four global nucleotide frequency parameters were used \cite{MuseGaut1994} and a parameterization which employed twelve nucleotide frequency parameters to allow for different frequencies at each codon position \cite{KosakovskyPondMuse2005}. We term the former framework MG1, and the latter MG3. Note that our MG1 corresponds to the original MG-style model \cite{MuseGaut1994}, whereas our MG3 corresponds to the so-called MG94$\times$HKY84 model \cite{KosakovskyPondMuse2005}. 

Figure~\ref{nyp_bias_r2} shows the resulting relationships between $dN/dS$ and $\omega$ MLEs for each set of mutation rates (NP, yeast, and polio), across model frequency parameterizations. Figure~\ref{nyp_bias_r2}A displays the estimator bias, the average discrepancy between the true $dN/dS$ value and the $\omega$ MLEs, of $\omega$ MLEs. Figure~\ref{nyp_bias_r2}B displays precision in this relationship, measured by the squared correlation coefficient $r^2$ between $dN/dS$ and $\omega$. The exact bias and $r^2$ values are given in Tables S1 and S2, respectively, and full regression plots for $\omega$ vs. $dN/dS$ are shown in Figure S1.

Two distinct trends emerge from Figure~\ref{nyp_bias_r2}. First, asymmetry in the mutational process consistently induced significant bias in $\omega$ estimates. Most often, the model underestimated $\omega$ relative to the true $dN/dS$ value. 
Based on simulations without any selection ($dN/dS = 1$) had previously suggested that GY-style models produce negatively biased $\omega$ estimates. By contrast, our results extend this finding and show that this bias is pervasive and remains approximately constant through a wide range of $dN/dS$ values (Figure~\ref{nyp_bias_r2}A, Table S1, Figure S1). Furthermore, we show that this bias systematically increased in magnitude as the underlying mutational process became more asymmetric. Indeed, for all frequency parameterizations, $\omega$ MLEs were most accurate under NP mutation rates, and both accuracy and precision tended to decrease as mutational bias progressed from yeast to polio mutation rates. Second, frequency parameterizations which more closely matched the mechanistic process that generated the data (MG1 and MG3) generally outperformed all other frequency estimators. In particular, MG1 clearly performed the best of all frequency estimators considered, featuring by far the least amount of estimator bias for the highly asymmetric polio mutation rates. Thus, going forward, we highly recommend that researchers employ MG-style matrix frameworks in their $dN/dS$ inferences to minimize bias.

Strikingly, when we examined model fit using AIC scores \cite{Akaike1974,BurnhamAnderson2004} for the different frequency parameterizations, we found that the F61 parameterization was unequivocally the best performing model, on average, for all datasets (Table~\ref{tab:dAIC}). This result dramatically juxtaposed the substantial inaccuracy and imprecision that F61 frequently yielded. In particular, F61 had the most estimator bias for NP datasets as well as the least precision for both NP and polio datasets (Figure~\ref{nyp_bias_r2}). Therefore, evaluating model performance based strictly on model fit can be highly misleading.  

That a standard measure of model fit (AIC) could not identify the model which produced the most accurate estimates for the parameter of interest has broad implications for model selection. In particular, model fit cannot necessarily identify the best-performing model if the data arose from a process distinct from the inference model. While certain models may have a better fit to the data at hand, this fit does not necessarily have any bearing on whether use of that model is mechanistically justified. In other words, model fit can be confounded with model accuracy, with regards to parameter estimates of interest.
Therefore, selecting models with metrics such as AIC may not guard effectively against spurious inferences. Instead, the mechanism producing the data should be carefully considered, and an appropriate inference method which best approximates this process should then be selected. Otherwise, model fit criteria may lend unfounded support to inappropriate models.


\section*{Conclusions}
By elucidating the relationship between $dN/dS$ and scaled selection coefficients, we have shown that $dN/dS$-based and MutSel models convey consistent information regarding the strength of natural selection. Importantly, our proof that $dN/dS \leq 1$ (assuming symmetric mutation rates and neutral synonymous changes) indicates that the use of MutSel models is only justified under conditions of strictly purifying selection. This restriction is in part indicated by the basic MutSel model assumption of constant selection pressures over time, or in other words a static fitness landscape \cite{HalpernBruno1998,Thorneetal2007,Rodrigueetal2010,Thorne2012}. Thus, if the aim is to identify positive selection, of the two frameworks examined here, only $dN/dS$-based models are appropriate. 


However, we also found that $dN/dS$ can readily be greater than 1 when selection acts on synonymous changes, even though the protein sequence is evolving solvely under purifying selection. This seemingly paradoxical finding actually reflected an assumption violation; the assertion that $dN/dS > 1$ necessarily corresponds to positive, diversifying selection requires that synonymous changes are neutral, which clearly does not hold if selection acts on synonymous changes. This result contributes to a growing body of literature which has found that purifying selection can yield $dN/dS > 1$ if model assumptions are not met. For instance, $dN/dS$ can theoretically be greater than 1, even under purifying selection, if sequences considered contain segregating polymorphisms \cite{Rochaetal2006,KryazhimskiyPlotkin2008,Mugaletal2014}. Thus, it is becoming increasingly clear that the $dN/dS = 1$ neutral threshold typically used to distinguish purifying and positive selection is highly sensitive to violations in model assumptions, and it is therefore critical to ensure that the  data adhere to model assumptions before conclusions from $dN/dS$ are drawn.

We suggest several strategies to limit such false positive results under synonymous selection. For one, certain formulations (particularly those implemented in HyPhy package \cite{KosakovskyPondetal2005}) of $dN/dS$-based methods model $dN$ and $dS$ rate variation separately \cite{MuseGaut1994,KosakovskyPondMuse2005,Murrell2013}, rather than using a single parameter to represent $dN/dS$. These sorts of methods may be able to identify when $dN/dS$ is greater than 1 not because of positive selection, but instead because $dS$ is vanishingly small due to strong synonymous selection. In addition, $dN/dS$-based models which have proposed corrections to $dS$ that account for synonymous selection may prove fruitful in the future \cite{Zhouetal2010}. Finally, our benchmarking approach, in which we simulate sequences according to MutSel models and infer $dN/dS$ both analytically and using ML, will help to identify circumstances under which synonymous selection confounds $dN/dS$ interpretations. 

Finally, we emphasize the utility of establishing relationships among distinct modeling frameworks in order to probe model behavior and evaluate model performance. Such an approach is uniquely able to reveal unrecognized behaviors and/or limitations of different modeling frameworks, and can precisely reveal the circumstances in which different models are best suited. We hope that further studies in this spirit will ensure robust model development going forward.


\section*{Methods}

\subsection*{Simulation of scaled selection coefficients.}

To examine the relationship between $dN/dS$ and scaled selection coefficients, we simulated 200 distributions of amino-acid scaled fitness values, $F^\text{aa}_a = 2Nf^\text{aa}_a$, from a normal distribution $\mathcal{N}(0,\sigma^2)$, where a unique $\sigma^2$ for each fitness distribution was drawn from a uniform distribution $\mathcal{U}(0,4)$. We converted these amino-acid fitnesses to codon fitnesses as follows. For 100 of the fitness distributions, we directly assigned all codons within a given amino acid family the fitness $F^\text{codon}_i = F^\text{aa}_a$, so that all synonymous codons had the same fitness. For the other 100 fitness distributions, we assigned synonymous codons different fitnesses by randomly selected a preferred codon for each amino acid. This preferred codon was assigned the fitness of $F^\text{codon}_i = F^\text{aa}_a + \lambda$, and all non-preferred codons were given the fitness $F^\text{codon}_i = F^\text{aa}_a - \lambda$. We drew a unique $\lambda$ for each fitness distribution from $\mathcal{U}(0,2)$. 
We then computed stationary codon frequencies as 
\begin{equation}\label{eq:boltzmann}
P_i = \frac{e^{F^\text{codon}_i}}{\sum e^{F^\text{codon}_k}}, 
\end{equation} where the sum in the denominator runs over all 61 sense codons \cite{SellaHirsh2005}. Equation \eqref{eq:boltzmann} gives the analytically precise stationary frequencies for a MutSel model, under the assumption of symmetric mutation rates \cite{SellaHirsh2005}. We used equations \eqref{eq:KN} - \eqref{eq:dS} to compute $dN/dS$ for each resulting set of stationary codon frequencies. For these calculations, we assumed the HKY85 \cite{HKY85} nucleotide mutation model, and accordingly we set the mutation rate for transitions as $\mu\kappa$, and the rate for all transversions as $\mu$. We used the value $\mu = 10^{-6}$ for all $dN/dS$ calculations, and we drew a unique value for $\kappa$ from $\mathcal{U}(1,6)$ for each set of codon frequencies.


\subsection*{Alignment simulations.}
We simulated protein-coding sequences as a continuous-time Markov process using standard methods \cite{Yang2006} according to the Halpern-Bruno MutSel model \cite{HalpernBruno1998}. In simplified form, this model's instantaneous rate matrix $Q = q_{ij}$ is populated by elements
\begin{equation}\label{eq:HBmatrix}
q_{ij} = \left\{ 
\begin{array}{rl}
m_{ij} \frac{S_{ij}}{1-1/S_{ij}} &\mbox{single nucleotide change} \\\\
0                                  &\mbox{multiple nucleotide changes} \\             
\end{array} \right.,
\end{equation} for a mutation from codon $i$ to $j$, where $m_{ij}$ is the mutation rate and the scaled selection coefficient $S_{ij}$ is defined in equation \eqref{eq:s_pmu}. All alignments presented here were simulated along a 4-taxon phylogeny (Figure~\ref{tree}), beginning with a root sequence generated in proportion to  stationary codon frequencies \cite{Yang2006}. Unless otherwise stated, all simulated alignments contained 500,000 codon positions. A single evolutionary model was applied to all positions in the simulated sequences. While this lack of site-wise heterogeneity is unrealistic for real sequence evolution, it allowed us to verify our derived relationship between scaled selection coefficients and $dN/dS$ with a sufficiently sized data set. 

\subsection*{Computation of stationary frequencies for experimental data sets.}
We used experimentally-determined site-specific amino-acid fitness parameters $F^\text{aa}_a$ for influenza nucleoprotein (NP), from ref.\ \cite{Bloom2014a}, in combination with experimental nucleotide mutation rates for either NP \cite{Bloom2014a}, yeast \cite{Zhu2014}, or polio virus \cite{Acevedo2014}, to derive realistic distributions of stationary codon frequencies. Ref.\ \cite{Bloom2014a} reported 498 distinct site-wise amino acid preference distributions for NP. We combined these 498 amino acid preference sets with each of the three mutation rate matrices sets to construct a total of $498 \times 3 = 1494$ unique experimental evolutionary Markov models, using the approach in refs.\ \cite{Bloom2014a,Bloom2014b}. The instantaneous rate matrix $Q$ for each experimental model is populated by elements
\begin{equation}
q_{ij} =  \left\{ 
\begin{array}{rl}
  F^\text{codon}_j \big{/} F^\text{codon}_i m_{ij} &\mbox{single nucleotide change, where $F^\text{codon}_j \geq F^\text{codon}_i$} \\
  m_{ij}                &\mbox{single nucleotide change, where $F^\text{codon}_j < F^\text{codon}_i$}  \\ 
  0                       &\mbox{multiple nucleotide changes} \\        
\end{array} \right.,
\end{equation} for a substitution from codon $i$ to codon $j$, where $F^\text{codon}_i$ is the fitness of codon $i$ \cite{Bloom2014a,Bloom2014b}. We calculated $F^\text{codon}_i$ values by simply assigning a given amino acid's experimental fitness $F^\text{aa}_a$ to each of its consistituent codons; thus, all synonymous changes were neutral. We determined the stationary codon frequencies for each resulting experimental model from the matrix's eigenvector corresponding to the eigenvalue 0. Finally, we simulated alignments for each set of stationary frequencies and corresponding mutation rates according to the Halpern-Bruno model (equation \eqref{eq:HBmatrix}).   

		
\subsection*{Maximum likelihood inference of dN/dS.}
For the 200 alignments simulated with symmetric mutation rates, we inferred $dN/dS$ using the M0 model \cite{Yangetal2000}, as implemented in the HyPhy batch language \cite{KosakovskyPondetal2005}. The M0 model uses the GY94 instantaneous rate matrix, which is populated by elements
\begin{equation}\label{eq:GY94}
q_{ij} = \left\{ 
	\begin{array}{rl}
	P_j                &\mbox{synonymous transversion} \\
	\kappa P_j           &\mbox{synonymous transition} \\
 	\omega P_j           &\mbox{nonsynonymous transversion} \\
 	\omega \kappa P_j    &\mbox{nonsynonymous transition} \\
	0                    &\mbox{multiple nucleotide changes} \\             
	\end{array} \right.,
\end{equation} for a substitution from codon $i$ to codon $j$, where $\kappa$ is the transition-transversion bias, $P_j$ is the equilibrium frequency of the target codon $j$, and $\omega$ represents $dN/dS$ \cite{GoldmanYang1994,NielsenYang1998}. The $P$ parameters are intended to represent those codon frequencies which would exist in absence of selection pressure generated by mutation alone \cite{GoldmanYang1994,MuseGaut1994,YN00,Yang2006}. Thus, when inferring $\omega$ on datasets which used symmetric mutation rates, we assigned the value $1/61$ to all parameters $P$, as all codons are  equally probable in the absence of mutational bias.

Alternatively, when inferring $\omega$ for alignments simulated with experimental fitness and mutation rates, we used several different model parameterizations, including GY-style \cite{GoldmanYang1994} (target codon frequency) and MG-style \cite{MuseGaut1994} (target nucleotide frequency) parameterizations. We considered the GY-style parameterizations F61 \cite{GoldmanYang1994}, F3x4 \cite{GoldmanYang1994}, CF3x4 \cite{KosakovskyPond2010}, and F1x4 \cite{MuseGaut1994}. We  implemented two varieties of MG-style models; the first, MG1, employs four parameters for nucleotide frequencies (one per nucleotide) and the second, MG3, employs twelve nucleotide frequency parameters, with four nucleotide frequency parameters for each of the three codon positions. All models included the parameters $\kappa$ and $\omega$. 

\subsection*{Availability}
All code is freely available from \texttt{https://github.com/clauswilke/Omega\_MutSel}, and simulated alignments are available from \textbf{dryad???}. Code used to simulate all alignments is available in a separate python library, at \texttt{https://github.com/sjspielman/pyvolve}.



\section*{Appendix 1}
We prove that $dN/dS \leq 1$ when calculated from scaled selection coefficients. We assume that mutation rates are symmetric ($m_{ij} = m_{ji}$) and that synonymous codons have the same fitness (synonymous changes are neutral). As described in the main text of this paper, these assumptions yield $dS = 1$, and hence we have to show that $dN = K_\text{N}\big{/}L_\text{N} \leq 1$. To this end, we note that the sums in $K_\text{N}$ and $L_\text{N}$ can be reordered such that the substitution probability from codon $i$ to $j$ is always added to the substitution probability from codon $j$ to $i$. We can then show that the sum of each of these pairs in the expression for $K_\text{N}$ is smaller than the term in $L_\text{N}$, and hence $dN/dS \leq 1$.

For this proof, we consider the pair of nonsynonymous codons $i$ and $j$, whose respective stationary frequencies $P_i$ and $P_j$ satisfy $P_i \leq P_j$ and $P_j > 0$. As follows from equations \eqref{eq:q_ij} and \eqref{eq:s_pmu}, the sum of the probability weights of evolving from codon $i$ to $j$ and from codon $j$ to $i$ is
\begin{equation}
N_em_{ij} u_{ij} + N_em_{ji} u_{ji}  = \frac{2P_iP_j[\log(P_i) - \log(P_j)]}{P_i - P_j} \,.
\end{equation}
This quantity represents $K_\text{N}$ in the $dN$ calculation. To prove $dN \leq 1$, we must show that this quantity is less than or equal to $P_i + P_j$, which represents $L_\text{N}$ in the $dN$ calculation. To this end, we introduce the function 
\begin{equation}\label{eq:Fxy}
F(x,y) = x + y - \frac{2xy[\log(x) - \log(y)]}{x - y} \,,
\end{equation}
and we will now show that $F(x,y) \geq 0$ for $x \leq y$ and $y \geq 0 $. It is straightforward to show, using l'H\^{o}pital's rule, that this condition holds for $x=y$. For $x < y$, we show that the first derivative of equation \eqref{eq:Fxy} is negative throughout $x \in (0,y)$, which proves that the function monotonically decreases, and thus $F(x,y) > 0$, in this interval. We calculate the first derivative as 
\begin{equation}
\frac{\partial{F(x,y)}}{\partial{x}} = \frac{\big{[} (x-3y)(x-y) - 2y^2(\log{x} - \log{y}) \big{]}}{(x-y)^2} \,.
\end{equation}
We now replace the expression $\log{x} - \log{y}$ by its Taylor expansion, yielding
\begin{equation}\label{eq:expand}
	\frac{\partial{F(x,y)}}{\partial{x}} = 
	\frac{ \bigg{[} (x-3y)(x-y) - 2y^2\bigg{(}\sum\limits_{n=1}^\infty \frac{1}{n} (1-x/y)^n\bigg{)} \bigg{]}}{(x-y)^2} \,.
\end{equation} We note that the first two terms of the Taylor series equal $(x-3y)(x-y)$, and thus expression \eqref{eq:expand} simplifies to 
\begin{equation}
\frac{\partial{F(x,y)}}{\partial{x}} = \frac{-2y^2\sum\limits_{n=3}^\infty \frac{1}{n} \big{(}1-\frac{x}{y}\big{)}^n}{(x-y)^2} ,
\end{equation}
which is clearly negative. This concludes the proof.


\bigskip



\section*{Appendix 2}
GY-style matrices may be expressed in the framework of the general-time reversible (GTR) model, in which the instantaneous matrix $Q$ can be decomposd into a $61 \times 61$ symmetric substitution rate matrix and a 61-dimensional vector containing the equilibrium codon frequencies, which correspond to the stationary distribution of the Markov chain. On the other hand, MG-style rate matrices explicitly consider nucleotide, not codon, frequencies, and thus do not clearly follow this paradigm. We now describe how the MG-style matrix can be rewritten in a form that follows the GTR framework. 
MG-style matrix elements, for a the substitution from codon $i$ to $j$, are generally given by 
\begin{equation}\label{eq:MGstyle}
Q_{ij} = \left\{ 
\begin{array}{rl}
\theta_{s_it_j}\pi_{t_j}          &\mbox{synonymous change} \\
\omega \theta_{s_it_j}\pi_{t_j}   &\mbox{nonsynonymous change} \\
0                             &\mbox{multiple nucleotide changes}       
\end{array} \right. ,
\end{equation} where $\omega$ is the ratio of nonsynonymous to synonymous substitution rates, and the product $\theta_{s_it_j}\pi_{t_j}$ corresponds to a nucleotide-level mutation rate $\mu_{s_it_j}$, where $s_i$ is the source nucleotide in codon $i$, and $t_j$ is the target nucleotide in codon $j$. This matrix yields the stationary codon frequency $P_i = \pi_{i_1}\pi_{i_2}\pi_{i_3}C$ for a given codon $i$, where $C = 1 - \Pi_\text{stop}$ and $\Pi_\text{stop} = \pi_\text{T}\pi_\text{A}\pi_\text{G} + \pi_\text{T}\pi_\text{G}\pi_\text{A} + \pi_\text{T}\pi_\text{A}\pi_\text{A}$ \cite{MuseGaut1994}. Therefore, we can simply rewrite the term $\theta_{s_it_j}\pi_{t_j}$ as $[P_j\theta_{s_it_j} / (\pi_m\pi_n)] \times C$, where $m$ and $n$ are the nucleotides which do not change in a given instantaneous codon substitution. This allows us to rewrite the rate instantaneous matrix as 
\begin{equation}
Q_{ij} = \left\{ 
\begin{array}{rl}
\frac{\theta_{s_it_j}}{\pi_m\pi_n}C P_j           &\mbox{synonymous change} \\\\
\omega\frac{\theta_{s_it_j}}{\pi_m\pi_n}C P_j     &\mbox{nonsynonymous change} \\\\
0                                              &\mbox{multiple nucleotide changes} \\           
\end{array} \right.
\end{equation} for a substitution from codon $i$ to codon $j$, and this matrix clearly conforms to the GTR framework.
		
%\begin{acknowledgments}
%	This work was supported by the army and by NIH. Computational resources were provided by CCBB.
% \end{acknowledgments}
		

		
		
\bibliographystyle{pnas}
\bibliography{bibliography}
		
		
%\end{article}
	
	
\section*{Figures and Tables}

\vspace{2cm}
	
\begin{figure}[htbp]
	\centerline{\includegraphics[width=8.7cm]{figures/MainText/dnds_variance.pdf}}
	\caption{\label{dnds_variance} $dN/dS$ decreass in proportion to amino-acid level selection strength. $dN/dS$ is plotted against the variance ($\sigma^2$) of the simulated distribution of amino-acid scaled selection coefficients. Higher variances indicate larger fitness differences among amino acids, whereas the limiting value of $\sigma^2 = 0$ indicates that all amino acids have the same fitness. (A) Synonymous codons have equal fitness values ($r^2=0.83$). (B) Synonymous codons have different fitness values ($r^2=0.45$). Note that panel B, but not A, shows $dN/dS$ values greater than 1, in spite of the steady-state evolutionary process.}
\end{figure}
		
		
\vspace{2cm}
		
\begin{figure}[htbp]
	\centerline{\includegraphics[width=8.7cm]{figures/MainText/regression_convergence.pdf}}
	\caption{\label{reg_conv} Combined modeling approach to assess performance of $dN/dS$ inference frameworks. (A) Protein-coding alignments are simulated in the MutSel modeling framework. $dN/dS$ can then be calculated (``predict") from scaled selection coefficients as well as through a ML inference framework (``infer"). Comparing resulting quantities reveals the accuracy of the chosen $dN/dS$ inference framework. (B) Regression between predicted $dN/dS$ values and inferred $\omega$ MLEs. Each point corresponds to a single simulated alignment, and the red line is the $x=y$ line. (C) Convergence of $\omega$ MLEs to the true $dN/dS$ value. The y-axis indicates the relative error of the $\omega$ MLE, and the x-axis indicates the number of positions in the simulated alignment. As the number of positions and hence the size of the data set increases, the $\omega$ converges to the $dN/dS$ values calculated using equations \eqref{eq:KN}-\eqref{eq:dS}. The red line is the $y=0$ line, indicating no error.}
\end{figure}
	
\vspace{2cm}
	

\begin{figure}[htbp]
	\centerline{\includegraphics[width=8.7cm]{figures/MainText/nyp_bias_r2.pdf}}
	\caption{\label{nyp_bias_r2} Estimator bias and precision of $\omega$ estimates for various model frequency parameterizations. (A) Estimator bias and (B) Precision ($r^2$) values between $dN/dS$ and $\omega$ MLEs across model frequency parameterizations, for each set of nucleotide mutation rates. To calculate bias, we fit a linear model with $\omega$ as the response and $dN/dS$ as the predictor, with a fixed slope of 1, and the resulting intercept value represents the bias. Negative biases indicate $\omega$ estimates that are, on average, lower than $dN/dS$. Note that all standard errors for bias are smaller than the symbol size.}	
\end{figure}

\vspace{2cm}


\begin{table}[htbp]
	\caption {\label{tab:dAIC} Mean $\Delta$AIC for datasets simulated with NP, yeast, or polio virus mutation rates.}
	\begin{tabular}{l c c c}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Frequencies} & NP & Yeast & Polio \\
		\noalign{\smallskip}\hline\noalign{\smallskip}
		F61 & 0 & 0 & 0 \\ 
		CF3x4 & -9627.5 & -7951.8 & -7975.9 \\ 
		MG1 & -13325.5 & -10042.0 & -5147.6 \\ 
		F1x4 & -13524.5 & -13658.5 & -15468.3 \\ 
		MG3 & -14401.3 & -12851.6 & -8624.9 \\ 
		F3x4 & -14807.2 & -17385.3 & -19384.6 \\ 
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	Based on AIC scores, the F61 parameterization strongly outperforms all other model parameterizations for all mutation rates, even though the F61 framework yields neither the most accurate nor the most precise parameter estimate. Note that the order of frequency models shown in the table corresponds to the model ranking for NP, and the ranking differs somewhat for yeast and polio datasets. AIC is computed as $AIC = 2(k - \ln L)$, where $k$ is the number of free parameters of the model, and $\ln L$ is the log-likelihood \cite{Akaike1974,BurnhamAnderson2004}. As codon and/or nucleotide frequency parameters were directly computed from the data, all models have 3 free parameters ($\omega$, $\kappa$, and a global branch-length scaling parameter).
\end{table}

\vspace{2cm}


\begin{figure}[htbp]
	\centerline{\includegraphics[width=6cm]{figures/MainText/simtree.pdf}}
	\caption{\label{tree} Phylogeny used for all simulated alignments.}	
\end{figure}


\clearpage

	
\section*{Supplementary Information}

\vspace{2cm}

\noindent Table S1. Estimator bias of $\omega$ MLEs and the true $dN/dS$ values, for all nucleotide mutation rates and model frequency parameterizations examined. All biases are statistically significant (different from 0), with all $P < 2\times10^{-16}$ except for the estimator bias associated with yeast mutation rates for MG3, where $P = 5.4\times10^{-5}$.
\begin{table}[htbp]
	\begin{tabular}{c c c c c c c}
		\hline\noalign{\smallskip}
		Mutation rate & MG1 & F1x4 & MG3 & CF3x4 & F3x4 & F61 \\
		\hline\noalign{\smallskip}
		NP & -0.014 & -0.02 & -0.007 & -0.009 & -0.007 & 0.019 \\ 
		Yeast & 0.025 & 0.007 & -0.063 & -0.084 & -0.076 & -0.068 \\ 
		Polio & -0.049 & -0.103 & -0.088 & -0.148 & -0.161 & -0.136 \\ 
		\noalign{\smallskip}\hline\noalign{\smallskip}
	\end{tabular}
\end{table}	


\vspace{2cm}

\noindent Table S2. Precision, measured as the squared correlation coefficient $r^2$, of $\omega$ MLEs relative to the true $dN/dS$ values, for all nucleotide mutation rates and model frequency parameterizations examined. All values shown are statistically significant, with all $P < 2\times10^{-16}$ .
\begin{table}[htbp]
	\begin{tabular}{c c c c c c c}
		\hline\noalign{\smallskip}
		Mutation rate & MG1 & F1x4 & MG3 & CF3x4 & F3x4 & F61 \\
		\hline\noalign{\smallskip}
		NP & 0.988 & 0.989 & 0.985 & 0.986 & 0.977 & 0.902 \\ 
		Yeast & 0.943 & 0.917 & 0.905 & 0.897 & 0.864 & 0.889 \\ 
		Polio & 0.842 & 0.811 & 0.777 & 0.754 & 0.781 & 0.752 \\ 
		\noalign{\smallskip}\hline\noalign{\smallskip}
	\end{tabular}
\end{table}	

\newpage

\begin{landscape}
	\includegraphics[width=9.5in]{figures/SI/nyp_regression.pdf}
	\vspace{0.5cm}
	
	Figure S1. Regressions of $\omega$ MLEs on the true $dN/dS$ values, as calculated from scaled selection coefficients, for datasets simulated using experimental fitnesses and mutation rates. Each point represents an alignment, and each red line is the $x=y$ line.
\end{landscape}



\end{document}

